src/main/java/enums/Currency.java

```diff
@@ -16,10 +19,20 @@ public enum Currency {
 
     public final String currencySymbol;
 
-    public static Currency fromSymbol(String symbol) {
-        return Arrays.stream(values())
-                .filter(currency -> currency.getCurrencySymbol().equals(symbol))
+    public static BigDecimal convertCurrency(String priceWithSymbol, Currency currency) {
+        var currencySymbol = currency.getCurrencySymbol();
+        return new BigDecimal(priceWithSymbol
+                .replace(currencySymbol, ""));
+    }
+
+    public static BigDecimal convertCurrencyFromCookie(String priceWithSymbol) {
```

Comment:
It can be called the same way as above (overload style)

Reply:
You are right! Renaming together with overload has simplified reading the code and calling these methods. 
```java
    public static BigDecimal getPriceAsBigDecimal(String priceWithSymbol, Currency currency) {
        var currencySymbol = currency.getCurrencySymbol();
        return new BigDecimal(priceWithSymbol
                .replace(currencySymbol, ""));
    }

    public static BigDecimal getPriceAsBigDecimal(String priceWithSymbol) {
        return getPriceAsBigDecimal(priceWithSymbol, getCookieManagerInstance().globalCurrency);
    }
```